subsys/bluetooth/host/hci_core.c

```diff
@@ -3271,7 +3271,7 @@ static void le_pkey_complete(struct net_buf *buf)
 	}
 
 	for (cb = pub_key_cb; cb; cb = cb->_next) {
-		cb->func(evt->status ? NULL : evt->key);
+		cb->func(evt->status ? NULL : pub_key);
```

Comment:
Actually, can you explain why evt->key is invalid? In the case that we got a zero status it should contain the valid new key. Could you also add this info to the commit message? (assuming I'm not missing something obvious here)

Reply:
Updated commit message and PR description.

Reply:
@joerchan that clarifies it, thanks!